engine-transactions/src/eip_1559.rs

```diff
@@ -70,7 +71,7 @@ pub struct SignedTransaction1559 {
 }
 
 impl SignedTransaction1559 {
-    pub fn sender(&self) -> Option<Address> {
+    pub fn sender(&self) -> Result<Address, ParseTransactionError> {
```

Comment:
I don't know if expanding `ParseTransactionError` to include variants for signature fails makes sense. The transaction has already been parsed into our data structure by this point. The question is whether the data that we parsed contains a valid signature or not. Maybe it makes sense to have a `SignatureVerificationError` type or something like that.

Reply:
Instead, I changed this to `Error` as its a crate Error. That also works.